pkg/generators/informergen/gen.go

```diff
@@ -0,0 +1,484 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package informergen
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"go/format"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/tools/go/packages"
+	"k8s.io/code-generator/cmd/client-gen/types"
+	"sigs.k8s.io/controller-tools/pkg/genall"
+	"sigs.k8s.io/controller-tools/pkg/loader"
+	"sigs.k8s.io/controller-tools/pkg/markers"
+
+	"github.com/kcp-dev/code-generator/pkg/flag"
+	"github.com/kcp-dev/code-generator/pkg/generators/clientgen"
+	"github.com/kcp-dev/code-generator/pkg/internal/informergen"
+	"github.com/kcp-dev/code-generator/pkg/util"
+)
+
+const (
+	// GeneratorName is the name of the generator.
+	GeneratorName = "informer"
+	// packageName for typed client wrappers.
+	typedPackageName = "externalversions"
+)
+
+type Generator struct {
+	// inputDir is the path where types are defined.
+	inputDir string
```

Comment:
The generator would run a few additional fields to find the right path to scaffold out the informers. https://github.com/kcp-dev/code-generator/blob/3290e34c024601662f2000893aa03361a5a15121/pkg/generators/clientgen/gen.go#L73-L80, something like this to consider the cases of having apis in subdirectories. If needed, please feel free to move  helpers like [these](https://github.com/kcp-dev/code-generator/blob/3290e34c024601662f2000893aa03361a5a15121/pkg/generators/clientgen/gen.go#L155-L162) out of client-gen.